lib/utils.coffee

```diff
@@ -70,9 +73,12 @@ exports.shouldUpdateToken = (token) ->
 ###
 exports.getAuthorizationHeader = Promise.method (token) ->
 	return if not token?
-	token.get().then (sessionToken) ->
-		return if not sessionToken?
-		return "Bearer #{sessionToken}"
+	token.hasKey().then (hasToken) ->
+		if !hasToken
+			return
+		token.getKey().then (sessionToken) ->
+			return if not sessionToken?
+			return "Bearer #{sessionToken}"
```

Comment:
All the code all the way through this library still says 'token', and it's mostly wrong (though probably not always).

Can you go through and rename the `token` variables to go along with this change? The `sessionToken` here for example is really just an unknown key - it might be a token, or it might be an api key, and we don't care which. This is especially important for places where `token` means the actual `resin-auth` instance, since that really doesn't make any sense. The `token` argument to `resin-request` should definitely be `auth` now.

@flesler did we settle on terminology for this? What's are we officially calling these? Do we have 'API key', and 'token', both of which are a type of 'key'?

Reply:
I used `Token` to refer in a general manner to both JWTs and API Keys and `Key` for the Token's string per se

Reply:
Some more debate: https://www.flowdock.com/app/rulemotion/resin-frontend/threads/lY3GgRC9VT07Crn7oRaNQrKauC-.

Conclusion: you authenticate with a 'token', which is either an 'api key', or a 'session token'.

That means:
* We should use be using `Token` in most places here to refer to the value (which we're mostly already doing, conveniently)
* `Key` refers to the actual string inside the token (as Ariel said above)
* The instantiated module should be called `auth`, because it's `resin-auth`
* If you have a case where you're only talking about one type specifically, it's either an 'api key', or a 'session token'